packages/javascript/bh-shared-ui/src/components/ExploreTable/ExploreTable.tsx

```diff
@@ -14,31 +14,111 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-import { Button } from '@bloodhoundenterprise/doodleui';
-import React from 'react';
-import useToggle from '../../hooks/useToggle';
-import { cn } from '../../utils/theme';
-import ManageColumns from './ManageColumns';
-
-interface ExploreTableProps {
-    open: boolean;
-    onClose: () => void;
+import { ColumnDef, DataTable } from '@bloodhoundenterprise/doodleui';
+import { faEllipsis } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { capitalize } from 'lodash';
+import { useMemo, useState } from 'react';
+import { makeFormattedObjectInfoFieldsMap } from '../../utils';
+import NodeIcon from '../NodeIcon';
+import { TableControls } from './TableControls';
+
+type HasData = { data?: object };
+
+interface ExploreTableProps<TData extends HasData> {
+    open?: boolean;
+    onClose?: () => void;
+    data?: Record<string, TData>;
 }
 
-const ExploreTable: React.FC<ExploreTableProps> = (props) => {
-    const { open, onClose } = props;
+const ExploreTable = <TData extends HasData>({ data, open, onClose }: ExploreTableProps<TData>) => {
+    const [searchInput, setSearchInput] = useState('');
+    const mungedData = useMemo(
+        () => (data && Object.keys(data).map((id) => ({ ...data?.[id]?.data, id }))) || [],
+        [data]
+    );
+
+    const firstItem = mungedData?.[0];
 
-    const [openManageColumns, toggleOpenManageColumns] = useToggle(false);
+    const labelsMap = makeFormattedObjectInfoFieldsMap(firstItem);
+
+    const initialColumns: ColumnDef<any, any>[] = [
+        {
+            accessorKey: '',
+            id: 'action-menu',
+            cell: () => (
+                <button className='pl-4'>
+                    <FontAwesomeIcon icon={faEllipsis} className='rotate-90 dark:text-neutral-light-1' />
+                </button>
+            ),
+        },
+        {
+            accessorKey: 'nonTierZeroPrincipal',
+            header: () => {
+                return <span className='dark:text-neutral-light-1'>Non Tier Zero Principal</span>;
+            },
+            cell: ({ row }) => {
+                return (
+                    <div className='flex justify-center items-center relative'>
+                        <NodeIcon nodeType={row?.original?.nodetype || 'N/A'} />
+                    </div>
+                );
+            },
+        },
+    ];
+
+    const columns: ColumnDef<any, any>[] = useMemo(
+        () =>
+            firstItem
+                ? // If column order exists in redux/localStorage, use that
+                  Object.keys(firstItem).map((key: any) => {
+                      return {
+                          accessorKey: key,
+                          header: labelsMap?.[key]?.label || capitalize(key),
+                          cell: (info: any) => String(info.getValue()),
```

Comment:
We should use the `format` util from `packages/javascript/bh-shared-ui/src/utils/entityInfoDisplay.ts` so that the cell values appear the same as they would in the entity panel!